### PR TITLE
Fix python meterpreter network method exceptions for OSX

### PR DESCRIPTION
**Pull request by adfoster-r7**
Fixes python meterpreter network method exceptions for OSX, and adds unit tests / CI automation

image

Fixes the following errors in [test/meterpreter](https://github.com/rapid7/metasploit-framework/blob/80e14846fe5feed734b3638899f0e2e11407a6c0/test/modules/post/test/meterpreter.rb) that I originally spotted running on CI

```diff
-            known_failures: [
-              [
-                [
-                  "[-] [should return network interfaces] FAILED: should return network interfaces",
-                  "[-] [should return network interfaces] Exception: Rex::Post::Meterpreter::RequestError : stdapi_net_config_get_interfaces: Operation failed: Python exception: TypeError",
-                  "[-] [should have an interface that matches session_host] FAILED: should have an interface that matches session_host",
-                  "[-] [should have an interface that matches session_host] Exception: Rex::Post::Meterpreter::RequestError : stdapi_net_config_get_interfaces: Operation failed: Python exception: TypeError"
-                ],
-                {
-                  if: [
-                    [
-                      :meterpreter_runtime_version,
-                      :==,
-                      "python3.8"
-                    ],
-                    :or,
-                    [
-                      :meterpreter_runtime_version,
-                      :==,
-                      "python3.6"
-                    ]
-                  ]
-                }
-              ]
-            ]
+            known_failures: []
```

### CODE REVIEW

File: python/meterpreter/README.md [hunk 1]
```diff
@@ -0,0 +1,7 @@
+## Python Meterpreter
+
+Running unit tests:
+
+```
+python -m unittest discover -v ./tests
```
Comment (adfoster-r7) [inline python/meterpreter/README.md:6]: I prefer pytest, but I thought only depending on the inbuilt testing library would be easier for contributors

File: python/meterpreter/ext_server_stdapi.py [hunk 2]
```diff
@@ -2188,8 +2188,8 @@ def stdapi_net_config_get_interfaces(request, response):
         iface_tlv += tlv_pack(TLV_TYPE_MAC_ADDRESS, iface_info.get('hw_addr', '\x00\x00\x00\x00\x00\x00'))
         if 'mtu' in iface_info:
             iface_tlv += tlv_pack(TLV_TYPE_INTERFACE_MTU, iface_info['mtu'])
-        if 'flags' in iface_info:
-            iface_tlv += tlv_pack(TLV_TYPE_INTERFACE_FLAGS, iface_info['flags'])
+        if 'flags_str' in iface_info:
+            iface_tlv += tlv_pack(TLV_TYPE_INTERFACE_FLAGS, iface_info['flags_str'])
```
Comment (adfoster-r7) [inline python/meterpreter/ext_server_stdapi.py:2192]: OSX was returning the integer flag value here, whilst linux was returning the string flag names. Now the implementations return both separately

File: python/meterpreter/ext_server_stdapi.py [hunk 3]
```diff
@@ -2524,7 +2526,7 @@ def stdapi_net_config_get_routes_via_osx_netstat():
             continue
         if destination == 'default':
             destination = all_nets
-        if re.match('link#\d+', gateway) or re.match('([0-9a-f]{1,2}:){5}[0-9a-f]{1,2}', gateway):
+        if re.match('link#\\d+', gateway) or re.match('([0-9a-f]{1,2}:){5}[0-9a-f]{1,2}', gateway):
```
Comment (adfoster-r7) [inline python/meterpreter/ext_server_stdapi.py:2529]: Fixes:
```
test_stdapi_net_config_get_interfaces_via_osx_ifconfig (test_ext_server_stdapi.TestExtServerStdApi.test_stdapi_net_config_get_interfaces_via_osx_ifconfig) ... <string>:2529: DeprecationWarning: invalid escape sequence '\d'
```

File: python/meterpreter/ext_server_stdapi.py [hunk 4]
```diff
@@ -2487,7 +2489,7 @@ def stdapi_net_config_get_routes_via_osx_netstat():
     proc_h = subprocess.Popen(['/usr/sbin/netstat', '-rn'], stdout=subprocess.PIPE, stderr=subprocess.PIPE)
```
Comment (adfoster-r7) [inline python/meterpreter/ext_server_stdapi.py:2489]: Looks like there's a file descriptor leak here, will put up a separate PR for this

```
$ python3 -m unittest discover -v ./tests
test_stdapi_net_config_get_interfaces (test_ext_server_stdapi.TestExtServerStdApi.test_stdapi_net_config_get_interfaces) ... <string>:2183: ResourceWarning: unclosed file <_io.BufferedReader name=3>
Object allocated at (most recent call last):
  File "/usr/local/Cellar/python@3.11/3.11.2_1/Frameworks/Python.framework/Versions/3.11/lib/python3.11/subprocess.py", lineno 1014
    self.stdout = io.open(c2pread, 'rb', bufsize)
<string>:2183: ResourceWarning: unclosed file <_io.BufferedReader name=5>
Object allocated at (most recent call last):
  File "/usr/local/Cellar/python@3.11/3.11.2_1/Frameworks/Python.framework/Versions/3.11/lib/python3.11/subprocess.py", lineno 1019
    self.stderr = io.open(errread, 'rb', bufsize)
```

File: python/meterpreter/tests/test_ext_server_stdapi.py [hunk 5]
```diff
@@ -0,0 +1,168 @@
...
+                "name": "en0",
+            }
+        ]
+
+        self.assertEqual(result, expected)
+
+    def test_stdapi_net_config_get_interfaces(self):
+        request = bytes()
+        response = bytes()
+        self.assertErrorSuccess("stdapi_net_config_get_interfaces", request, response)
```
Comment (adfoster-r7) [inline python/meterpreter/tests/test_ext_server_stdapi.py:149]: I'll skip verifying the exact bytes for now, this was just an MVP "make sure it doesn't explode" test for now

File: python/meterpreter/ext_server_stdapi.py (hunk 2 above)
Comment (jmartin-r7) [reply python/meterpreter/ext_server_stdapi.py:2192]: This seems to be introducing new data into the TLV, is there value for both vs just shifting to a more consistent type retuned in `flags`?

Comment (adfoster-r7) [reply python/meterpreter/ext_server_stdapi.py:2192]: `TLV_TYPE_INTERFACE_FLAGS` has always been a string:

```
TLV_TYPE_INTERFACE_FLAGS    = TLV_META_TYPE_STRING  | 1403
```

Previously the OSX code path was was extracting out the integer value for flags instead of a string like in the Linux codepath, i.e. in the osx code path attempted to write out `8836` instead of `"UP,BROADCAST,SMART,RUNNING,SIMPLEX,MULTICAST"` from the below ifconfig output

```
en5: flags=8863<UP,BROADCAST,SMART,RUNNING,SIMPLEX,MULTICAST> mtu 1500
```

This is now updated to send the string flag names for both OSX and Linux

Comment (jmartin-r7) [reply python/meterpreter/ext_server_stdapi.py:2192]: Got it, `stdapi_net_config_get_interfaces_via_netlink` was returning `flags` as a set of strings names and  `stdapi_net_config_get_interfaces_via_osx_ifconfig` was returning `flags` as numeric values now

Both return `flags` as numeric values and `flags_str` as string names and only `flags_str` is being placed in the response as a `TLV_TYPE_INTERFACE_FLAGS` packet.

File: python/meterpreter/README.md [hunk 6]
```diff
@@ -0,0 +1,11 @@
+## Python Meterpreter
+
+Running unit tests:
+
+```
+# Only required if less than python 3.3
+pip install mock
```
Comment (dwelch-r7) [inline python/meterpreter/README.md:7]:
```suggestion
python -m pip install mock
```
